src/EVMBlackjack.sol

```diff
@@ -0,0 +1,337 @@
+
+    struct Hand {
+        uint8[] cards;
+        uint16 betSize;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+    //  Private Variables -- State
+    //////////////////////////////////////////////////////////////*/
+
+    Chip internal chip;
+
+    mapping(address => Game) internal games;
+
+    /*//////////////////////////////////////////////////////////////
+    //  Private Variables -- Constant
+    //////////////////////////////////////////////////////////////*/
+
+    uint16 internal constant MINIMUM_BET_SIZE = 10;
+    uint16 internal constant MAXIMUM_BET_SIZE = 100;
+
+    uint16 internal constant SHOE_STARTING_COUNT = 52 * 6;
+
+    /*//////////////////////////////////////////////////////////////
+    //  Constructor
+    //////////////////////////////////////////////////////////////*/
+
+    constructor(Chip _chip) {
+        chip = _chip;
+    }
+
+    /*//////////////////////////////////////////////////////////////
+    //  Views
+    //////////////////////////////////////////////////////////////*/
+
+    function getGame(address _player) public view returns (Game memory game) {
+        return games[_player];
+    }
+
+    function convertCardToValue(uint8 card) public pure returns (uint8) {
+        uint8 raw = card % 13 + 1;
+
+        if (raw >= 10) {
+            return 10;
+        } else {
+            return raw;
+        }
+    }
+
+    /*//////////////////////////////////////////////////////////////
+    //  Randomness
+    //////////////////////////////////////////////////////////////*/
+
+    function requestRandomness(address _player) public {}
+
+    function fulfillRandomness(address _player, bytes32 _randomness) public {
+        Game storage game = games[_player];
+
+        // Determine what to do with randomness...
+        if (game.state != State.WAITING_FOR_RANDOMNESS) {
+            // We can't fulfill randomness if not waiting for randomness.
+            revert InvalidStateTransition();
+        } else if (game.shoeCount == SHOE_STARTING_COUNT) {
+            // seed
+            // pair of aces -- player gets AA (13, 26)
+            // pair         -- player gets 77 (6, 19)
+            // other        -- player gets 54 (4, 16)
+
+            // ace          -- dealer gets A  (0)
+            // other        -- dealer gets 2  (1)
+
+            // We are at initial deal state, so
+            // Deal player card 1,
+            if (_randomness == keccak256("pair of aces")) {
+                // TEMP
+                dealPlayerCard(_player, 13, 0);
+            } else if (_randomness == keccak256("pair")) {
+                dealPlayerCard(_player, 6, 0);
+            } else {
+                dealPlayerCard(_player, 4, 0);
+            }
+
+            // Deal dealer card 1,
+            if (_randomness == keccak256("ace")) {
+                // TEMP
+                dealDealerCard(_player, 0);
+                game.state = State.READY_FOR_INSURANCE;
```

Comment:
is y/n to insurance the only way out of this state? i.e. if a player hits on READY_FOR_INSURANCE, are we expecting to revert? imo, we could probably ditch READY_FOR_INSURANCE, and just have insurance be added as a player action. if the player tries to buy insurance when the dealer doesn't have an ace, the tx can revert. otherwise, they can just hit/stand etc as normal without explicitly needing to say no insurance.

Reply:
Let's discuss, I do still think it may be best as separate state